src/main/java/flashcard/FlashcardDeck.java

```diff
@@ -56,4 +56,26 @@ public void testRandomCard() {
             }
         }
     }
+
+    public void deleteCard() {
+        Scanner in = new Scanner(System.in);
+        System.out.println("Please enter the index of the card you wish to delete.");
+        String userInput = in.nextLine();
+        int cardIndex = 0;
+        try {
+            cardIndex = Integer.parseInt(userInput);
+        } catch (NumberFormatException e) {
+            System.out.println("Please enter the index of the card as an integer!");
+        }
+        if (cardIndex > flashcardDeck.size()) {
+            System.out.println("Sorry, you only have " + flashcardDeck.size() + " cards in your deck!\n"
+                    + "Please enter a number within the range of 1-" + flashcardDeck.size() + ".");
+        } else {
+            assert flashcardDeck.size() <= cardIndex : "size should not be greater than card index at this step";
```

Comment:
Your error: java.lang.AssertionError: size should not be greater than card index at this step

I think you meant assert flashcardDeck.size() >= cardIndex

Reply:
I think it'll be clearer if you put

```
assert cardIndex <= flashcardDeck.size() : "card index inserted should be less than size of deck at this step"
```
assert is used to check if the value of something is what it **should actually be**, in this case, you want to assert that card index is actually less than or equal to deck size.

Consider adding another check for cardIndex to be greater than 0 btw! Otherwise negative values might break the application.